### PR TITLE
Port some code from vue3 migration into master and reduce the amount of entry code

### PR DESCRIPTION
**Pull request by codyrancher**
### Summary
Port some code from vue3 migration into master and reduce the amount of entry code

### Areas or cases that should be tested
Any page, harvester more likely than other pages.

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/initialize/app-extended.js [hunk 1]
```diff
@@ -102,20 +101,6 @@ async function extendApp(vueApp) {
         resolve();
       });
     });
-
-    router.afterEach((to) => {
```
Comment (codyrancher) [inline shell/initialize/app-extended.js:106]: Moved to navigation guard.

File: shell/initialize/entry-helpers.js [hunk 2]
```diff
@@ -53,232 +47,25 @@ export const loadDebugger = (vueApp) => {
   }
 };
 
-/**
- * Handle errors with a redirect
- * @param {*} context
- * @param {*} message
- */
-const errorRedirect = (context, message) => {
```
Comment (codyrancher) [inline shell/initialize/entry-helpers.js:61]: Not needed after deleting other code

File: shell/initialize/entry-helpers.js [hunk 3]
```diff
@@ -53,232 +47,25 @@ export const loadDebugger = (vueApp) => {
...
-  context.$router.replace('/fail-whale');
-};
-
-/**
- * TODO: Define this logic use case
- * @param {*} fn
- * @param {*} context
- * @returns
- */
-export const promisify = (fn, context) => {
```
Comment (codyrancher) [inline shell/initialize/entry-helpers.js:72]: Not needed after deleting other code

File: shell/initialize/entry-helpers.js [hunk 4]
```diff
@@ -53,232 +47,25 @@ export const loadDebugger = (vueApp) => {
...
 
-/**
- * Render function used by the router guards
- * @param {*} to Route
- * @param {*} from Route
- * @param {*} next callback
- * @param {*} app
- * @returns
- */
-async function render(to, from, next) {
```
Comment (codyrancher) [inline shell/initialize/entry-helpers.js:109]: I slowly removed more and more sections of this unitl I realized it wasn't adding value any longer. 

Ultimately the biggest thing this did was call `mount()` (passed to it inside mountApp) after it resolved. This wasn't very clear until I removed more of the implementation.

File: shell/initialize/entry-helpers.js [hunk 5]
```diff
@@ -53,232 +47,25 @@ export const loadDebugger = (vueApp) => {
...
-  const mount = () => {
-    vueApp.$mount('#app');
-  };
-
   // Initialize error handler
   vueApp.$loading = {}; // To avoid error while vueApp.$nuxt does not exist
 
-  // Add beforeEach router hooks
-  router.beforeEach(render.bind(vueApp));
-  router.afterEach((from, to) => {
```
Comment (codyrancher) [inline shell/initialize/entry-helpers.js:218]: Moved to navigation guard

File: shell/initialize/entry-helpers.js [hunk 6]
```diff
@@ -53,232 +47,25 @@ export const loadDebugger = (vueApp) => {
...
-    if (from?.name !== to?.name) {
-      updatePageTitle(getVendor());
-    }
-  });
-
-  // First render on client-side
-  const clientFirstMount = () => {
```
Comment (codyrancher) [inline shell/initialize/entry-helpers.js:225]: Wrapper of a wrapper

File: shell/initialize/entry-helpers.js [hunk 7]
```diff
@@ -289,65 +76,6 @@ export const getMatchedComponents = (route, matches = false, prop = 'components'
   }));
 };
 
-const getComponent = async(unknownComponent) => {
```
Comment (codyrancher) [inline shell/initialize/entry-helpers.js:292]: Unused after other code was deleted

File: shell/initialize/entry-helpers.js [hunk 8]
```diff
@@ -289,65 +76,6 @@ export const getMatchedComponents = (route, matches = false, prop = 'components'
...
-
-  return componentView || unknownComponent;
-};
-
-/**
- * Patch all the matched components of a given route
- * @param {*} route
- * @returns
- */
-const patchMatchedComponents = (route) => Array.prototype.concat.apply(
```
Comment (codyrancher) [inline shell/initialize/entry-helpers.js:330]: Unused after other code was deleted

File: shell/initialize/entry-helpers.js [hunk 9]
```diff
@@ -380,59 +121,13 @@ export const setContext = async(app, context) => {
...
       isDev:   true,
-      isHMR:   false,
       app,
       store:   app.store,
       payload: context.payload,
       error:   context.error,
       base:    app.router.options.base,
-      env:     {
-        commit: 'head', version: '0.1.2', dev: true, pl: 1, perfTest: false, rancherEnv: 'web', api: 'http://localhost:8989'
-      }
```
Comment (codyrancher) [inline shell/initialize/entry-helpers.js:391]: These are completely unused and replaced by the usage of DefinePlugin inside vue.config.js.

**Comment by codyrancher** [comment]
Some basic verification of harvester:

https://github.com/user-attachments/assets/3d606768-88d1-4bcf-a7a5-887b4c498892

File: shell/initialize/entry-helpers.js (hunk 4 above)
Comment (rak-phillip) [reply shell/initialize/entry-helpers.js:109]: This is a great removal - `render()` contains a lot of complexity and has traditionally been the source of a great deal of confusion in trying to understand the "why" of its existence.

**Review by rak-phillip — COMMENTED** [review]
All the changes represented here look great. This simplifies our init a great deal, which is something that we absolutely need.

I ran this through some smoke tests and most everything looks great. However, I am able to reproduce an error that gets printed to the console while logging out and logging back in:

```
[Vue warn]: Failed to mount component: template or render function not defined. found in ---> <Index> at shell/pages/index.vue <Root>
```

### Repro steps
- Log into Dashboard
- Log out of Dashboard
- Log into Dashboard again
- Observe the console

I tried the repro steps in v.2.9.0 environment and I'm not able to produce the same behavior.

**Review by rak-phillip — APPROVED** [review]
Actually, I'm able to reproduce this error in a development environment in master

```
[Vue warn]: Failed to mount component: template or render function not defined.

found in

---> <Shell/pages/index.vue> at shell/pages/index.vue
       <Root>
```

I don't think this error is related to PR at all; I'll follow up with an issue so we can investigate to root cause.

File: shell/initialize/entry-helpers.js (hunk 4 above)
Comment (cnotv) [reply shell/initialize/entry-helpers.js:109]: Oh yes please, I was agonizing with this one

**Pull request merged**